### PR TITLE
[google_maps_flutter_platform_interface] Add a `fromJson` constructor.

### PR DESCRIPTION
**Pull request by ditman**
## Description

Add a `fromJson` constructor to `BitmapDescriptor`.

This is required so serialized descriptors can be synchronously re-hydrated in the Web implementation.

## Related Issues

Needed for: https://github.com/flutter/flutter/issues/66622

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart [hunk 1]
```diff
@@ -116,6 +116,12 @@ class BitmapDescriptor {
     return BitmapDescriptor._(<dynamic>['fromBytes', byteData]);
   }
 
+  /// The inverse of .toJson.
+  /// Do not use this method, it can be removed at any time.
+  /// This is needed in Web to re-hydrate BitmapDescriptors that have been
+  /// transformed to JSON for transport.
+  BitmapDescriptor.fromJson(dynamic json) : _json = json;
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart:123]: Can we add a TODO to link to an issue for removing this method?
Also, does it make sense to mark this as deprecated if we know we are going to remove it?

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart:123]: Yes, I'll add a TODO to migrate the API of the google_maps_flutter plugin and link it here. It does make sense to be @Deprecated('Will be removed with: LINK_TO_THE_ISSUE').

I'll change this right away.

Comment (cyanglaz) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart:123]: cc @amirh 
Is it ok to add deprecation notes on a temporary API that we just add.

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart:123]: This is the crux of the issue: https://github.com/flutter/flutter/issues/70330

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart:126]: We discussed this offline. I'll add some validation to the fromJson method to try to ensure that users can't create objects substantially different than those by the static methods, and remove the deprecation notice.

The bigger problem here is that the API is currently promising to return a JSON object with certain format, and not really the way it's created.

**Review by cyanglaz — APPROVED** [review]
LGTM!

**Pull request merged**